### PR TITLE
Add more test cases for writable external table

### PR DESCRIPTION
**Pull request by adam8157**
The old cdbfast test suite had these, move to ICW.

Signed-off-by: Adam Lee <ali@pivotal.io>
Signed-off-by: Yuan Zhao <yuzhao@pivotal.io>

### CODE REVIEW

File: src/bin/gpfdist/regress/output/exttab1.source [hunk 1]
```diff
@@ -553,6 +553,19 @@ SELECT DISTINCT(R_NAME) FROM ret_region ORDER BY R_NAME;
...
 -- start_ignore
+DROP TABLE IF EXISTS table_multi_locations;
+NOTICE:  table "table_multi_locations" does not exist, skipping
+DROP EXTERNAL TABLE IF EXISTS ext_table_multi_locations;
+NOTICE:  table "ext_table_multi_locations" does not exist, skipping
+-- end_ignore
+-- The use of more than one gpfdist. Each gpfdist points to a different file to write into
+CREATE TABLE table_multi_locations (a int, b text) DISTRIBUTED BY (a);
+INSERT INTO table_multi_locations VALUES (generate_series(1,256), 'test_1');
+CREATE WRITABLE EXTERNAL TABLE ext_table_multi_locations (a int, b text) LOCATION ('gpfdist://ccle-gpdb:7070/wet_multi_locations_1.tbl', 'gpfdist://ccle-gpdb:7070/wet_multi_locations_2.tbl') FORMAT 'TEXT' (DELIMITER AS '|' NULL AS 'null');
```
Comment (hlinnaka) [inline src/bin/gpfdist/regress/output/exttab1.source]: Need to use @hostname@ here.

File: src/test/regress/input/external_table.source [hunk 2]
```diff
@@ -1360,6 +1369,93 @@ DROP EXTERNAL TABLE IF EXISTS exttab_with_option_1;
...
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_csv3;
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_csv4;
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_csv5;
+-- end_ignore
+
+-- Create writable external table with AS for DELIMITER , NULL, ESCAPE
+CREATE WRITABLE EXTERNAL TABLE tbl_wet_csv1 (a int, b text) LOCATION ('gpfdist://@hostname@/wet_csv.tbl') FORMAT 'CSV' (DELIMITER AS '|' NULL AS 'null' ESCAPE AS ' ');
+
+-- Create writable external table without AS for DELIMITER , NULL, ESCAPE
+CREATE WRITABLE EXTERNAL TABLE tbl_wet_csv2 (a int, b text) LOCATION ('gpfdist://@hostname@/wet_csv.tbl') FORMAT 'CSV' (DELIMITER AS ',' NULL 'null' ESCAPE ' ');
```
Comment (hlinnaka) [inline src/test/regress/input/external_table.source]: I don't think we need to test the same thing both with and without AS. It's handled very early in the grammar.

It makes sense to use AS in some tests, but not in others, so that we have some coverage for both, but we don't really need dedicated tests for it.

File: src/test/regress/input/external_table.source [hunk 3]
```diff
@@ -1360,6 +1369,93 @@ DROP EXTERNAL TABLE IF EXISTS exttab_with_option_1;
...
+
+-- Create writable external table with double quotes
+CREATE WRITABLE EXTERNAL TABLE tbl_wet_csv3 (a int, b text) LOCATION ('gpfdist://@hostname@/wet_csv.tbl') FORMAT 'CSV' (DELIMITER AS '|' NULL AS 'null' ESCAPE AS ' '  QUOTE AS '"');
```
Comment (hlinnaka) [inline src/test/regress/input/external_table.source]: These tests would be a lot more interesting if they actually wrote something to the external tables. As is, these only really exercise the parser. I'd suggest using file:/// protocol for these, and actually write something.

Note that there are similar tests, albeit for readable external tables, in gpfdist's exttab1 test.

File: src/test/regress/input/external_table.source [hunk 4]
```diff
@@ -1360,6 +1369,93 @@ DROP EXTERNAL TABLE IF EXISTS exttab_with_option_1;
...
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_syntax1;
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_syntax2;
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_syntax3;
+
+-- start_ignore
+DROP TABLE IF EXISTS table_execute;
+DROP EXTERNAL TABLE IF EXISTS tbl_wet_execute;
+-- end_ignore
+
+CREATE TABLE table_execute (id integer, name varchar(40) CONSTRAINT con1 CHECK (id > 99 AND name <> '')) DISTRIBUTED RANDOMLY;
```
Comment (hlinnaka) [inline src/test/regress/input/external_table.source]: The CHECK constraint seems irrelevant here.

Comment (hlinnaka) [inline src/test/regress/input/external_table.source]: Or is the CHECK constraints copied to the external table, with the LIKE? But it doesn't do anything to verify that it's enforced. Or is it testing that it does *not* get copied?

Comment (adam8157) [inline src/test/regress/input/external_table.source]: This case in cdbfast was just to test writable external web table using `execute`, will remote the CHECK, thanks for reviewing

File: src/test/regress/input/external_table.source (hunk 3 above)
Comment (adam8157) [inline src/test/regress/input/external_table.source]: `file://` protocol doesn't support writing, use web table here, thanks.
